src/lib.rs

```diff
@@ -113,25 +172,29 @@ where
 
         match bucket.take() {
             Bucket::Empty => {
-                *bucket = Bucket::Leaf(KvPair(key, val));
+                *bucket = Bucket::Leaf(KvPair { key, val });
                 Ok(None)
             }
-            Bucket::Leaf(KvPair(old_key, old_val)) => {
+            Bucket::Leaf(KvPair {
+                key: old_key,
+                val: old_val,
+            }) => {
                 if key == old_key {
-                    *bucket = Bucket::Leaf(KvPair(key, val));
+                    *bucket = Bucket::Leaf(KvPair { key, val });
                     Ok(Some(old_val))
                 } else {
                     let mut new_node = Hamt::new();
                     let old_hash = Id::new(&old_key).hash();
 
                     new_node._insert(key, val, hash, depth + 1)?;
                     new_node._insert(old_key, old_val, old_hash, depth + 1)?;
-                    *bucket = Bucket::Node(Annotated::new(new_node));
+                    *bucket = Bucket::Node(Link::new(new_node));
                     Ok(None)
                 }
             }
             Bucket::Node(mut node) => {
-                let result = node.val_mut()?._insert(key, val, hash, depth + 1);
+                let result =
+                    node.compound_mut()?._insert(key, val, hash, depth + 1);
```

Comment:
The same question of the other PR, why the naming for this `insert` includes the underscore as prefix?